runtime/integration-tests/src/generic/cases/lp/pool_management.rs

```diff
@@ -0,0 +1,249 @@
+
+	// Verify the  test currencies are correctly added to the pool manager
+	assert_eq!(
+		Decoder::<H160>::decode(
+			&env.view(
+				Keyring::Alice,
+				"pool_manager",
+				"currencyIdToAddress",
+				Some(&[Token::Uint(Uint::from(index.index))])
+			)
+			.unwrap()
+			.value
+		),
+		test_erc20_address
+	);
+
+	assert_eq!(
+		Decoder::<Balance>::decode(
+			&env.view(
+				Keyring::Alice,
+				"pool_manager",
+				"currencyAddressToId",
+				Some(&[Token::Address(test_erc20_address)]),
+			)
+			.unwrap()
+			.value
+		),
+		index.index
+	);
+}
+
+fn add_pool<T: Runtime>() {
+	let mut env = super::setup::<T>(|_| {});
+	const POOL: PoolId = 1;
+
+	env.parachain_state_mut(|| {
+		crate::generic::utils::pool::create_one_tranched::<T>(
+			Keyring::Admin.into(),
+			POOL,
+			LocalUSDC::ID,
+		);
+
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_pool(
+			OriginFor::<T>::signed(Keyring::Admin.into()),
+			POOL,
+			Domain::EVM(EVM_DOMAIN_CHAIN_ID)
+		));
+
+		utils::process_outbound::<T>()
+	});
+
+	let creation_time = env.parachain_state(<pallet_timestamp::Pallet<T> as TimeAsSecs>::now);
+
+	// Compare the pool.created_at field that is returned
+	assert_eq!(
+		Decoder::<Uint>::decode(
+			&env.view(
+				Keyring::Alice,
+				"pool_manager",
+				"pools",
+				Some(&[Token::Uint(Uint::from(POOL))]),
+			)
+			.unwrap()
+			.value
+		),
+		Uint::from(creation_time)
+	);
+
+	env.pass(Blocks::ByNumber(1));
+
+	env.parachain_state_mut(|| {
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_pool(
+			OriginFor::<T>::signed(Keyring::Admin.into()),
+			POOL,
+			Domain::EVM(EVM_DOMAIN_CHAIN_ID)
+		));
+
+		utils::process_outbound::<T>()
+	});
+
+	// TODO: Actually find the revert event here.
+	//
+	// NOTE: That is really relevant for the router too.
+	//     We need to check `Pending` and see for errors
+	//     unfortuntately and then error in the router instead of returning success.
+
+	// Adding a pool again DOES NOT change creation time - i.e. not override storage
+	assert_eq!(
+		Decoder::<Uint>::decode(
+			&env.view(
+				Keyring::Alice,
+				"pool_manager",
+				"pools",
+				Some(&[Token::Uint(Uint::from(POOL))]),
+			)
+			.unwrap()
+			.value
+		),
+		Uint::from(creation_time)
+	);
+}
+
+fn add_tranche<T: Runtime>() {
+	let mut env = super::setup::<T>(|env| {
+		super::setup_currencies(env);
+		super::setup_pools(env);
+	});
+
+	env.parachain_state_mut(|| {
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_tranche(
+			OriginFor::<T>::signed(Keyring::Admin.into()),
+			POOL_A,
+			pallet_pool_system::Pallet::<T>::pool(POOL_A)
+				.unwrap()
+				.tranches
+				.ids
+				.last()
+				.unwrap()
+				.clone(),
+			Domain::EVM(EVM_DOMAIN_CHAIN_ID)
+		));
+
+		utils::process_outbound::<T>()
+	});
+}
+
+fn allow_investment_currency<T: Runtime>() {
+	let mut env = super::setup::<T>(|env| {
+		super::setup_currencies(env);
+		super::setup_pools(env);
+		super::setup_tranches(env);
+	});
```

Comment:
Missing the purpose of this call

```suggestion
fn allow_investment_currency<T: Runtime>() {
	let mut env = super::setup::<T>(|env| {
		super::setup_currencies(env);
		super::setup_pools(env);
		super::setup_tranches(env);
	});
	todo!("allow investment currency");
```

Reply:
Not sure what you mean?

Reply:
If I am not mistaken here, the current implementation only includes the setup but not the actual allowing of the investment currency. I am trying to locate the missing pieces of this PR. Please correct me if I am wrong about any of the pinged functions, i.e. if you believe they are complete in the current state.